File: app/src/main/java/com/smartscanner/service/SummarizerResultStore.java
package com.smartscanner.service;

import androidx.annotation.Nullable;

public final class SummarizerResultStore {
    @Nullable
    public static volatile String lastResult = null;
    public static volatile boolean isSuccess = false;
    public static volatile boolean isLoading = false;

    private SummarizerResultStore() {
    }

    public static void clear() {
        lastResult = null;
        isSuccess = false;
        isLoading = false;
    }
}

File: app/src/main/java/com/smartscanner/service/SummarizerResultStore.kt
package com.smartscanner.service

object SummarizerResultStore {
    @Volatile var lastResult: String? = null
    @Volatile var isSuccess: Boolean = false
    @Volatile var isLoading: Boolean = false

    fun clear() {
        lastResult = null
        isSuccess = false
        isLoading = false
    }
}
